### DISCUSSION TITLE
Is it possible to use sherpa with XSPEC models without installing local HEASoft?

### DISCUSSION THREAD

**Discussion by swagner-astro** [sherpa/sherpa#1837] [Q&A]
Hi there,

I am new to X-ray analysis and have only worked on a remote machine in XSPEC so far. Now I would like to migrate to my own machine (macOS Monterey 12.6.2) and work in python. However, I do want to use XSPEC models, specifically cflux. From what I read online this means I have to have a local version of the HEASOFT Software. Is there any way to do so without installing that for the sake of storage on my machine?

(More Info about my setup) 
At the moment I have installed sherpa in a separate conda environment via `conda install -c sherpa sherpa` and I get this warning and error: 
```
>from sherpa.astro.ui import *
WARNING: failed to import sherpa.astro.xspec; XSPEC models will not be available
>mymod = xspowerlaw.mypow
ImportError: cannot import name 'xspec' from 'sherpa.astro'
```

In the [documentation](https://sherpa.readthedocs.io/en/latest/install.html#building-from-source) I see I would have to build sherpa from source and have a local HEADAS installation (the $HEADAS value must be replaced by the actual path to the HEADAS installation). I am getting stuck in that process and before digging further into that, I wanted to ask if there is any workaround. 
Thank you!

**Comment by hamogu** [comment]
The easiest for you is probably to install a pre-build Sherpa that already comes with XSPEC. Both are part of the CIAO installation, see https://cxc.cfa.harvard.edu/ciao/download/

IF you take the commands from that page, you get all the XSPEC models, but you also get CIAO and the CALDB. IF you want to save space, you can get just sherpa with XSPEC and the necessary dependencies (e.g. Python, matplotlib, ...). I think that should be:
```
$ conda create -n ciao-4.15-sherpa \
  -c https://cxc.cfa.harvard.edu/conda/ciao \
  -c conda-forge sherpa python=3.10
```

There is no way to use the XSPEC models in a local installation without having them installed locally in some way (either through conda or by compiling them manually).

**Comment by DougBurke** [comment]
So, if you really only want the cflux model then you can look at https://cxc.harvard.edu/ciao/guides/sherpa-flux.html since we have other ways to calculate the flux. Or you (or we) can create an equivalent convolution-style model to replicate cflux.

However, I would be surprised if you need to calculate fluxes but not use an XSPEC model.
